### DISCUSSION TITLE
none-ls and luasnip completion

### DISCUSSION THREAD

**Discussion by ikirudennis** [nvimtools/none-ls.nvim#130] [General]
I've been a longtime vim user, and have been working on migrating my setup over to neovim. I followed some youtube tutorials on how to get lsp and completion working, but I ran into some trouble with trying to enable the none-ls builtin completion for luasnip.

I spent some time digging into this, and I eventually found that this line: https://github.com/nvimtools/none-ls.nvim/blob/37671797d6e5f9bd0ee3ab965ab2e6de251e1930/lua/null-ls/builtins/completion/luasnip.lua#L47 seems to be the culprit. The snippet description is a table (a table containing only a string, incidentally), and when nvim-cmp tries to work with it, it's expecting `detail` to be a string. I also tried looking into the file history, but the line linked above pre-dates luasnip 1.0.0, so it's hard to say what `detail` or `description` were really intended to do. I found that either commenting out the linked line or changing it to `snip.description[1]` was effective at getting none-ls to work as a completion system. But ultimately, I found it to be redunant, and luasnip was actually providing snippets properly, so I just removed `null_ls.builtins.completion.luasnip` from my config.

This raises a few questions:
- Which project is wrong here? none-ls, luasnip, or nvim-cmp? 
- If none-ls is the project that should fix it, which is the right solution? Commenting out the line, changing it to `snip.description[1]` or maybe hard-coding the detail to `"none-ls luasnip completion"` so a user can at least know which completion system is producing the given snippet (This is one change I tried to confirm which snippet system was causing trouble)? The description is included in the documentation, so it's a bit redundant.
- Is `null_ls.builtins.comnpletion.luasnip` even needed anymore?

**Comment by mochaaP** [comment]
You don't need `null_ls.builtins.completion.luasnip` if LuaSnip already provided a source for your completion plugin.

This is only useful when your completion plugin doesn't support LuaSnip (e.g. mini.complete).

**Reply by ikirudennis** [reply]
Aha! Thanks for that info. Is this something I overlooked in the docs? Or, should none-ls check for the existence nvim-cmp and print out a warning message and stop loading or something?
